actions/wait-for-workflow-job/action.yml

```diff
@@ -0,0 +1,137 @@
+        RETRY_DELAY: ${{ inputs.retry-delay }}
+        TIMEOUT: ${{ inputs.timeout }}
+      run: |
+        # Convert sleep_time to an integer if necessary (Bash typically handles this, but it's good practice)
+        sleep_time=$(($RETRY_DELAY + 0))
+        # Set a maximum wait time
+        max_wait_seconds=$(($TIMEOUT + 0))
+        start_time=$(date +%s)
+
+        # wait until the job starts and then grab its job id, or until we timeout
+        while :; do
+            current_time=$(date +%s) # Get the current time in seconds since epoch
+            elapsed_time=$((current_time - start_time))
+
+            # Check if the maximum wait time has been exceeded
+            if [ "$elapsed_time" -ge "$max_wait_seconds" ]; then
+                echo "Timeout reached without job starting."
+                exit 1
+            fi
+
+            # Fetch the list of jobs from the GitHub API
+            jobs=$(curl -H "Authorization: token $GH_TOKEN" \
+              -H "Accept: application/vnd.github.v3+json" \
+              "https://api.github.com/repos/${REPO}/actions/runs/$WF_RUN_ID/jobs?per_page=100")
+
+            # Extract the total count of jobs, defaulting to '0' if not present
+            job_count=$(echo "$jobs" | jq '.total_count // 0')
+            if [ "$job_count" -eq 0 ]; then
+              echo "No jobs found yet. Sleeping $sleep_time seconds and then trying again..."
+              sleep $sleep_time
+              continue
+            fi
+
+            # Extract the job ID for the specified job name
+            job_id=$(echo "$jobs" | jq --arg job_name "$JOB_NAME" -r '.jobs[] | select(.name == $job_name) | .id')
+
+            # Check if the job ID is found
+            if [ -n "$job_id" ]; then
+                echo "Job exists."
+                break
+            fi
+
+            echo "Sleeping for $sleep_time seconds before checking for the job again..."
+            sleep $sleep_time
+        done
+
+        # If the job ID is still not found, exit with an error
+        if [ -z "$job_id" ]; then
+            echo "Job '${JOB_NAME}' was never found."
+            exit 1
+        fi
+
+        # Set a maximum wait time (e.g., 30 minutes) in seconds
+        max_wait_seconds=$((30 * 60))
+        start_time=$(date +%s)
+
+        # Loop until the job completes, or we timeout
+        while :; do
+            current_time=$(date +%s)
+            elapsed_time=$((current_time - start_time))
+
+            # Check if the maximum wait time has been exceeded
+            if [ "$elapsed_time" -ge "$max_wait_seconds" ]; then
+                echo "Timeout reached without job starting."
+                break
+            fi
+
+            # Fetch the job details from the GitHub API
+            job_details=$(curl -H "Authorization: token $GH_TOKEN" \
+                -H "Accept: application/vnd.github.v3+json" \
+                "https://api.github.com/repos/${REPO}/actions/jobs/$job_id")
```

Comment:
Same as above.

Reply:
hmm I just read this here:

`This command does not support authenticating via fine grained PATs as it is not currently possible to create a PAT with the checks:read permission.`

https://cli.github.com/manual/gh_run_view

does this not affect us?

Reply:
Now uses gh cli: https://github.com/smartcontractkit/.github/pull/454/commits/e02e6b59110bbf3ca6c14a23d1fa1641d2b0bde0